main/src/library/Fixpoint3/Ram.flix

```diff
@@ -0,0 +1,493 @@
+/*
+ * Copyright 2021 Benjamin Dahse
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *                Adam Yasser Tallouzi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+mod Fixpoint3.Ast.Ram {
+    // We have not updated these imports to be for Fixpoint3 as we have not changed any of the files
+    use Fixpoint.Ast.Shared.{BoxedDenotation => Denotation, PredSym}
+    use Fixpoint.Ast.Shared.Denotation.{Relational}
+    use Fixpoint.Boxed
+    use Fixpoint.PredSymsOf
+    use Fixpoint.SubstitutePredSym
+
+    /////////////////////////////////////////////////////////////////////////////
+    // RamProgram                                                              //
+    /////////////////////////////////////////////////////////////////////////////
+
+    ///
+    /// A `RamProgram` consists of a statements, the EDB, predicate information,
+    /// and the information about the indexes to be constructed.
+    /// 
+    pub enum RamProgram {
+        case Program(RamStmt, Facts, Predicates, IndexInformation)
+    }
+
+    ///
+    /// `Predicate` represents 1 + the maximal indentifier of the full `RelSym`s of the program and set of `RelSym`
```

Comment:
You mention a program here but there is no program in the type, so maybe just delete that part?